src/components/signup/SignUp.js

```diff
@@ -0,0 +1,181 @@
+import React, { Component } from 'react';
+import logo from '../../images/logo.png';
+import { Link } from 'react-router-dom';
+import './SignUp.css';
+
+const emailRegex = RegExp(/^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/);
+//Add the telephone regex
+const formValid = ({ formErrors, ...rest }) => {
+  let valid = true;
+
+  // validate form errors being empty
+  Object.values(formErrors).forEach((val) => {
+    val.length > 0 && (valid = false);
+  });
+
+  // validate the form was filled out
+  Object.values(rest).forEach((val) => {
+    val === null && (valid = false);
+  });
+
+  return valid;
+};
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      fullName: null,
+      email: null,
+      password: null,
+      phoneNumber: null,
+      confirmPassword: null,
+      formErrors: {
+        fullName: '',
+        email: '',
+        phoneNumber: '',
+        password: '',
+        confirmPassword: ''
+      }
+    };
+  }
+  handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (formValid(this.state)) {
+      console.log(`
```

Comment:
@morelmiles the console logs from line 45 - 56 are good for testing but let's not push those to develop, whenever you're done, you can always clean those up

Reply:
i have updated this